frontend/components/ifixit-image/ifixit-image.tsx

```diff
@@ -33,3 +34,59 @@ function isStrapiImage(src: string) {
       /^https:\/\/ifixit-(dev-)?strapi-uploads.s3.amazonaws.com\//
    );
 }
+
+const guideImageLoader: ImageLoader = ({
+   src,
+   width,
+   quality,
+}: ImageLoaderProps) => {
+   const baseSrc = src.replace(/\.[^/.]+$/, '');
+   const sizeName = getImageSize(width, guideImageSizeMap, 'huge');
+   // We don't use the ?width param server-side, but it gets rid of a nextjs warning
+   return baseSrc.concat('.', sizeName, `?width=${width}`);
+};
+
+const cartImageLoader: ImageLoader = ({
+   src,
+   width,
+   quality,
+}: ImageLoaderProps) => {
+   const baseSrc = src.replace(/\.[^/.]+$/, '');
+   const sizeName = getImageSize(width, cartImageSizeMap, 'size1000');
+   return baseSrc.concat('.', sizeName, `?width=${width}`);
+};
```

Comment:
I think we could have something like `getImageLoader` that returns a function that produces an image loader. Maybe something like:

```ts
function getImageLoader(sizeMap: SizeMap, defaultSize: string): ImageLoader {
   return ({ src, width }: ImageLoaderProps) => {
      const baseSrc = src.replace(/\.[^/.]+$/, '');
      const sizeName = getImageSize(width, sizeMap, defaultSize);
      return baseSrc.concat('.', sizeName, `?width=${width}`);
   };
}
```